src/main/java/org/jasig/portlet/calendar/dao/HibernateCalendarSetDao.java

```diff
@@ -54,18 +59,21 @@ public void setCalendarStore(CalendarStore calendarStore) {
      * @see org.jasig.portlet.calendar.dao.ICalendarSetDao#getCalendarSet(javax.portlet.PortletRequest)
      */
     public CalendarSet<?> getCalendarSet(PortletRequest request) {
-        
-        final String username = getUsername(request);
-                
-        // retrieve a list of calendar configurations for this user
-        final List<UserDefinedCalendarConfiguration> cals = calendarStore
-            .getCalendarConfigurations(username);
-        
+
         final Set<UserDefinedCalendarConfiguration> calendars = new HashSet<UserDefinedCalendarConfiguration>();
-        calendars.addAll(cals);
-        
-        final CalendarSet<UserDefinedCalendarConfiguration> set = new CalendarSet<UserDefinedCalendarConfiguration>();
-        set.setConfigurations(calendars);
+        final CalendarSet<UserDefinedCalendarConfiguration> set
+                = new CalendarSet<UserDefinedCalendarConfiguration>(calendars);
+
+        final String username = getUsername(request);
+        if (username != null) {
```

Comment:
At this point, the session init code should have set this to 'guest' if the user is unauthenticated. Yep, logging is warranted.